### ISSUE TITLE
Deepin 15.8 ... System Tray ... Suggestion ... to simplify process work flow ...

### ISSUE THREAD

**Issue by aiamuzz** [linuxdeepin/dde-dock#128]
Hi,

I just upgraded to Deepin 15.8 and have been test driving it ...

I noticed there is a new 'System Tray' icon in place of the earlier circle with the systray icons ... this implementation is good ...

Wouldn't it be better that instead of one having to click on the system tray icon to open the system tray to access the applications there ... a more efficient way would be to let the system tray open itself say after a 2 sec mouse hover on it ?

the 'mouse hover' function is already in place for other icons in the dock where the application name pop's up ... could this same mouse hover functions on the 'System Tray' Icon trigger the opening of the system tray exposing the applications inside ?

Further this System Tray Icon can shut back after a few seconds of inactivity ... this way the aesthetics of the dock will be retained ... as the current System Tray in an open/drawn state is not very appealing visually.

**Another suggestion is you might consider dropping the rounded square background for the system tray application icons ... and revert back to the system tray implementation found in 'Efficient Mode' which does not have these background boxes.**

**Comment by es20490446e** [comment]
I simply feel it's better to have all the icons always at view. But only have icons that are constantly used, and put the rest of functions on the Control Center only.

Included:
🔉 Sound
🔋 Battery

Excluded:
🖧 Wired network
📶 Wireless network
🕑 Time
🔌 Shutdown

**Comment by es20490446e** [comment]
For time management it's better to set an alarm than watching the time constantly.

For shutting down the computer it's more straightforward from the physical button. Alternatively also having the button on the Control Center.

**Comment by es20490446e** [comment]
Merging network icons can also be an option, as that from the user perspective what's relevant is if they are connected. They probably already know what network interface they are using.

**Comment by es20490446e** [comment]
The Disk icon probably is better as a big icon, so to prevent less experienced users to disconnect devices without them noticing the dangers.

For that purpose naming it as "safely remove" would be more meaningful for them.

**Comment by aiamuzz** [comment]
> I simply feel it's better to have all the icons always at view.

@es20490446e  ... in efficient(windows panel) mode ... always in view may work up to a point as there is enough space ...

but it will fail miserably in the Dock(Fashion) mode ... 

1. the icons need to be of the same size if they are kept always in view cluttering the dock unnecessarily

2. 'System Tray icon' is a good approach in Dock(fashion) mode the  ...
only the cumbersome clicking is a little annoying ... hence the request/suggestion of opening the system tray on 'Mouse Hover' action ... and defaulting this icon to a closed state after a certain time has lapsed after its opened ...

**Comment by es20490446e** [comment]
I'm going to say this as someone who likes interfaces to be as clean as possible. Also thinking more about the average user, rather than my own personal taste.

If you condense the tray, you cannot notice right away the sound volume, the battery level, if the network is connected, or if the file transmission software is still working. Condensing defeats the warning functionality of the tray.

If you have all the icons of the same size it becomes noticeably harder to distinguish which are running applications, quick access applications, background services and plugins. Changing among them becomes noticeably more confusing.

So I believe the best approach is having only the icons that are really useful in that place, the ones that need to be used constantly.

**Comment by aiamuzz** [comment]
@es20490446e ...

> If you have all the icons of the same size it becomes noticeably harder to distinguish which are running applications, quick access applications, background services and plugins. Changing among them becomes noticeably more confusing.

Best you switch to efficient mode ... you can't hope to have all that on a dock ... 

As a professional designer i can say for sure, what you are hoping is going to be a mess visually on a Dock mode ... the dock HAS to have equally sized icons if there are to be placed side by side else its a mess ... look how messy an open systray measures in comparison to the other icons on the dock ... to keep the aesthetics of dock intact its best to close the systray after about 5 seconds !!!

**Comment by es20490446e** [comment]
I also extensively know about UX design too, and software engineering.You probably know my work: the Ubuntu desktop origami look and feel, and the Ubuntu papercuts quality processes.

I have tried all the Linux desktops that exist, and I have extensively read and experimented about UX design and software design in general. I have coded in nine different programming languages. I'm trained in quality assurance in general, and process engineering. I have mentored in Google events, and worked in well know software companies.

So we simply have different opinions, not a matter of who is more qualified :stuck_out_tongue: In the end of the day it isn't about who has the best curriculum, but about which idea works better. It can come from the most genius person, or from mommy. Great ideas that I implemented came from computer illiterate people.

What I'm simply stating is this, that these are the principles that worked for me:
- Decide on the most important first: functionality first, aesthetics later. Form follows function.
- Show what is used in this very moment, or constantly, and hide everything else.
- What is used is useful, what isn't used is useless. If people normally use the tray, and nothing else has proven by demonstration to do the work better, it shall be included.
